src/Compiler/Checking/Expressions/CheckComputationExpressions.fs

```diff
@@ -849,6 +850,23 @@ let (|OptionalSequential|) e =
     | SynExpr.Sequential(debugPoint = _sp; isTrueSeq = true; expr1 = dataComp1; expr2 = dataComp2) -> (dataComp1, Some dataComp2)
     | _ -> (e, None)
 
+//   use! x = ...
+//   use! (x) = ...
+//   use! (__) = ...
+//   use! _ = ...
+//   use! (_) = ...
+[<return: Struct>]
+let rec (|UnwrapUseBang|_|) supportsUseBangBindingValueDiscard pat =
```

Comment:
Is the function name fitting?
It does not have anything to do with `use` nor `bang`, does it?

Isn't it really simplifying a complex pattern into an identifier  extracted out of that pattern?

Reply:
Yeah we can remove this and just get the ident and pat where we needed it.